File: RecipeManager/RecipeManager/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RecipeManager.Models
{
    public class Product
    {
        public int Id { get; set; }
		private string _name;
		public string Name
		{
			get { return _name; }
			set
			{
				_name = value.ToLower();
			}
		}
		public ProductUnit Unit { get; set; }
        public double Count { get; set; }
        public bool IsInFridge { get; set; }

        public Recipe Recipe { get; set; }
        public int? RecipeId { get; set; }
        public Shopping Shopping { get; set; }
        public int? ShoppingId { get; set; }
		public Fridge Fridge { get; set; }
		public int? FridgeId { get; set; }
	}
}

File: RecipeManagerSPA/src/app/_models/product.ts
export interface Product {
    name: string;
    unit: number;
    count: number;
}

export interface ProductEdit {
    id: number;
    name: string;
    unit: number;
    count: number;
}

export interface ShoppingProduct {
    name: string;
    unit: number;
    count: number;
    boughtCount: number;
    fridgeCount: number;
    isInFridge: boolean;
}
